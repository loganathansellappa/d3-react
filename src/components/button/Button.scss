.button-container {
  display: flex;
  gap: 5px;
  margin-bottom: 10px;
  position: sticky;
  .button {
    border-radius: 8px;
    border: 1px solid transparent;
    padding: 0.6em 1.2em;
    font-size: 1em;
    font-weight: 500;
    font-family: inherit;
    cursor: pointer;
    transition: border-color 0.25s;
    color: #000;
  }

  .button-active {
    background-color: #007bff;
    color: #ffffff;
  }

  button:hover {
    border-color: #646cff;
  }
  button:focus,
  button:focus-visible {
    outline: 4px auto -webkit-focus-ring-color;
  }
}
