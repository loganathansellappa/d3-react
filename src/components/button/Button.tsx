import React, {useCallback, useState} from "react";
import "./Button.scss";
interface ButtonProps {
    buttonOneClick: () => void;
    buttonTwoClick: () => void;
    labelOne: string;
    labelTwo: string;
}

export const Button: React.FC<ButtonProps> = ({ buttonOneClick, buttonTwoClick, labelOne, labelTwo }) => {
    const [isButtonOneActive, setIsButtonOneActive] = useState(true);

    const handleButtonOneClick = useCallback(() => {
        buttonOneClick();
        setIsButtonOneActive(true);
    }, [buttonOneClick]);

    const handleButtonTwoClick = useCallback(() => {
        buttonTwoClick();
        setIsButtonOneActive(false);
    }, [buttonTwoClick]);

    return (
        <>
            <div className={"button-container"}>
                <button           className={`button ${isButtonOneActive ? "button-active" : ""}`}
                                   onClick={handleButtonOneClick}>
                    {labelOne}
                </button>
                <button          className={`button ${!isButtonOneActive ? "button-active" : ""}`}
                                 onClick={handleButtonTwoClick}>
                    {labelTwo}
                </button>
            </div>
        </>

    );
};
