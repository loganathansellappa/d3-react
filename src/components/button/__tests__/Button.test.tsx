import { fireEvent, render } from "@testing-library/react";
import { Button } from "../Button";

describe("Button", () => {
  let element: { container: HTMLElement };
  const buttonProps = {
    buttonOneClick: jest.fn(),
    buttonTwoClick: jest.fn(),
    labelOne: "labelOne",
    labelTwo: "labelTwo",
  };
  beforeEach(() => {
    element = render(<Button {...buttonProps} />);
  });

  test("renders Button component", () => {
    expect(element.container.firstChild).toMatchSnapshot();
  });

  test("displays button label", () => {
    expect(element.container.textContent).toContain("labelOne");
    expect(element.container.textContent).toContain("labelTwo");
  });

  test("handles button one click event", () => {
    const buttonOne = element.container.getElementsByClassName("button")[0];
    fireEvent.click(buttonOne);
    expect(buttonProps.buttonOneClick).toHaveBeenCalled();
  });

  test("handles button two click event", () => {
    const buttonTwo = element.container.getElementsByClassName("button")[1];
    fireEvent.click(buttonTwo);
    expect(buttonProps.buttonTwoClick).toHaveBeenCalled();
  });
});
