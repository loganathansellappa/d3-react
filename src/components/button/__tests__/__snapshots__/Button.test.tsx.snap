// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Button component 1`] = `
<div
  class="button-container"
>
  <button
    class="button button-active"
  >
    labelOne
  </button>
  <button
    class="button "
  >
    labelTwo
  </button>
</div>
`;
